crates/egui/src/containers/panel.rs

```diff
@@ -50,44 +50,205 @@ impl PanelState {
 
 // ----------------------------------------------------------------------------
 
-/// [`Left`](Side::Left) or [`Right`](Side::Right)
+/// [`Left`](VerticalSide::Left) or [`Right`](VerticalSide::Right)
 #[derive(Clone, Copy, Debug, PartialEq, Eq)]
-pub enum Side {
+pub enum VerticalSide {
     Left,
     Right,
 }
 
-impl Side {
+/// [`Top`](HorizontalSide::Top) or [`Bottom`](HorizontalSide::Bottom)
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum HorizontalSide {
+    Top,
+    Bottom,
+}
+
+/// [`Horizontal`](PanelSide::Horizontal) or [`Vertical`](PanelSide::Vertical)
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum PanelSide {
+    Vertical(VerticalSide),
+    Horizontal(HorizontalSide),
+}
+
+impl PanelSide {
     fn opposite(self) -> Self {
+        let opposite_vertical = |side: VerticalSide| -> PanelSide {
+            match side {
+                VerticalSide::Left => PanelSide::Vertical(VerticalSide::Right),
+                VerticalSide::Right => PanelSide::Vertical(VerticalSide::Left),
+            }
+        };
+
+        let opposite_horizontal = |side: HorizontalSide| -> PanelSide {
+            match side {
+                HorizontalSide::Top => PanelSide::Horizontal(HorizontalSide::Bottom),
+                HorizontalSide::Bottom => PanelSide::Horizontal(HorizontalSide::Top),
+            }
+        };
```

Comment:
I think all the above would look a lot more readable as member methods:

`side.opposite()` (for all three side enum types)